### PR TITLE
csh wrapper

### PR DESCRIPTION
**Pull request by kalefranz**
Basic wrapper for csh.

### CODE REVIEW

**Comment by kalefranz** [comment]
Let's get https://github.com/conda/conda/pull/6446 merged first

**Comment by kalefranz** [comment]
Right now, I believe csh is mostly working, even though the integration test doesn't pass for it.  Integration test passes for tcsh.  Because of that integration test issue, I don't yet think we're done with the csh/tcsh story.  But this PR is a good start that lays out the framework.

File: shell/etc/csh/login.d/conda.csh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+# Recommended way to make the conda command available in csh is
+#   $ sudo ln -s <CONDA_ROOT>/etc/csh/login.d/conda.sh /etc/csh/login.d/conda.sh
```
Comment (nehaljwani) [inline shell/etc/csh/login.d/conda.csh]:
```
s/conda.sh/conda.csh
```
I am not sure, but shouldn't this be `/etc/profile.d/conda.csh` ? That's where `/etc/csh.login` sources stuff from.

Comment (kalefranz) [reply shell/etc/csh/login.d/conda.csh]: Really? Must be different on every distribution. That sucks. I think I looked at csh on Ubuntu. Guess I’ll check a couple different distros and see if I can tease out what’s most common

File: conda/activate.py [hunk 2]
```diff
@@ -165,8 +173,9 @@ class Help(CondaError):  # NOQA
             from .exceptions import ArgumentError
             raise ArgumentError("invalid command '%s'" % command)
         elif command == 'activate' and len(remainder_args) > 1:
+            print(sys.argv, file=sys.stderr)
```
Comment (nehaljwani) [inline conda/activate.py]: Is this deliberate or a debug statement?

Comment (kalefranz) [reply conda/activate.py]: yeah there's a problem here in this block

File: shell/etc/csh/login.d/conda.csh (hunk 1 above)
Comment (kalefranz) [reply shell/etc/csh/login.d/conda.csh]: Ok, on Ubuntu, `csh` doesn't pre-populate any files in `/etc`, but `tcsh` does.  Maybe that's the difference I'm seeing...

Comment (kalefranz) [reply shell/etc/csh/login.d/conda.csh]:
```
root@47cf55bec20a:/# apt-get install csh
...
root@47cf55bec20a:/# echo "echo hello world" > /etc/profile.d/hello.csh
root@47cf55bec20a:/# . !$
. /etc/profile.d/hello.csh
hello world
root@47cf55bec20a:/# csh
#
```

I guess that didn't work.

**Comment by nehaljwani** [comment]
This functionality for csh is not 100% complete yet, but I'm still going to go ahead and merge this.
The part missing here is that `$prompt` is not exported by csh (at least in my case), so it is not available to python. So after activating an environment, only the modifier is displayed and on deactivation, nothing is displayed.
```
[nwani@centos7-nwani ~]$ sudo docker exec -it 519cb4b3763f /bin/csh
[root@519cb4b3763f /build_scripts]# source /opt/conda/etc/profile.d/conda.csh
[root@519cb4b3763f /build_scripts]# conda activate root
(base) echo $prompt
(base)
(base) conda deactivate

echo $prompt

set prompt="[%n@%m %c]%#"
[root@519cb4b3763f /build_scripts]#
```

**Comment by kalefranz** [comment]
Agreed. The prompt modifier is not yet complete. At least we have a start though.

**Pull request merged**